inyoka_theme_default/templates/portal/privmsg/index.html

```diff
@@ -0,0 +1,209 @@
+  {{ macros.breadcrumb_item(_('Portal'), href('portal', 'privmsg', 'inbox')) }}
+{% endblock %}
+
+{% block content %}
+  {%- if message %}
+    <div class="message">
+      <div class="head">
+        <div class="row">
+          <div class="col-sm-9">
+            <dl>
+              <dt>{% trans %}From:{% endtrans %}</dt>
+              <dd>
+                <a{% if not message.author.is_active %} class="user_inactive"{% endif %} href="{{ message.author|url }}">
+                  {{- message.author|e }}
+                </a>
+              </dd>
+              <dt>{% trans %}To:{% endtrans %}</dt>
+              <dd>
+                {% for user in message.recipients -%}
+                  <a{% if not user.is_active %} class="user_inactive"{% endif %} href="{{ user|url }}">{{ user|e }}</a>
+                  {%- if not loop.last %}, {% endif %}
+                {% else %}
+                  …
+                {% endfor %}
+              </dd>
+              <dt>{% trans %}Subject:{% endtrans %}</dt>
+              <dd>{{ message.subject|e }}</dd>
+              <dt>{% trans %}Date:{% endtrans %}</dt>
+              <dd>{{ message.pub_date|datetime }}</dd>
+            </dl>
+          </div>
+          <div class="col-sm-3 avatar">{{ macros.add_user_avatar(message.author) }}</div>
+        </div>
+      </div>      
+      <div class="body">
+        {{ message.rendered_text }}
+      </div>
+      <div class="foot">
+        {%- if not message.author == USER %}
+          {% if message.author.is_active %}
+            <a class="pn_reply" href="{{ message|url('reply') }}"><i class="fa_icon-reply"></i> <span>{% trans %}Reply{% endtrans %}</span></a> |
+          {%- endif %}
+
+          {%- if message.recipients|length > 1 %}
+            <a class="pn_reply_all" href="{{ message|url('reply_to_all') }}"><i class="fa_icon-reply-all"></i> <span>{% trans %}Reply to all{% endtrans %}</span></a> |
+          {%- endif %}
+        {%- endif %}
+        <a class="pn_forward" href="{{ message|url('forward') }}"><i class="fa_icon-share"></i> <span>{% trans %}Forward{% endtrans %}</span></a> |
+        {%- if folder.id == 'trash' %}
+          <a class="pn_restore" href="?action=restore"><i class="fa_icon-arrow-up"></i> <span>{% trans %}Restore{% endtrans %}</span></a> |
+        {%- elif folder.id != 'archive' %}
+          <a class="pn_archive" href="?action=archive"><i class="fa_icon-archive"></i> <span>{% trans %}Archive{% endtrans %}</span></a> |
+        {%- endif %}
+        <a class="pn_delete" href="?action=delete"><i class="fa_icon-trash-o"></i> <span>{% trans %}Delete message{% endtrans %}</span></a>
+      </div>
+      <div class="spam-info">
+        {%- trans moderators='http://wiki.ubuntuusers.de/ubuntuusers/Moderatoren' -%}
+          If you received a spam message, please forward it to a <a href="{{ moderators }}">moderator</a>.
+        {%- endtrans -%}
+      </div>
+    </div>
+  {%- endif %}
+
+  <form action="" method="post">
```

Comment:
form-macros will not give (a lot of) reduction of the source code.  However, it would be more consistent to use them here, too, imho.